commons/zenoh-codec/src/core/zint.rs

```diff
@@ -134,18 +144,20 @@ where
 
     fn read(self, reader: &mut R) -> Result<u64, Self::Error> {
         let mut b = reader.read_u8()?;
+        if (b & 0x80) == 0 {
```

Comment:
do we really need this piece of code, given that even without it the result would be the same ?

Reply:
Yes, it's an early return that avoids useless shift and improves the performance for common case when a `zint` is smaller than `128`.

Reply:
Ok, if it is indeed the case. I just thought that an extra if branch might have a higher impact overall than an extra bitshift.

Reply:
Counterintuitively that's not the case.